core/src/ledger_cleanup_service.rs

```diff
@@ -60,8 +60,10 @@ impl LedgerCleanupService {
 
         let (slot, _) = slot_full_receiver.recv_timeout(Duration::from_secs(1))?;
         if slot > max_ledger_slots {
+            let last = slot - max_ledger_slots;
             //cleanup
-            blocktree.purge_slots(0, Some(slot - max_ledger_slots));
+            blocktree.purge_slots(0, Some(last));
+            blocktree.compact_storage(0, last)?;
```

Comment:
I'd move this into purge, purge should mean fully delete.

Reply:
great idea! fixed in latest